spec/api/v1/product_show_controller.rb

```diff
@@ -0,0 +1,5 @@
+require 'rails-helper'
+
+RSpec.describe Api::V1::ProductShowController, type: :controller do
```

Comment:
On line 3 it references a `ProductShowController`, but I only see a `ProductsController`.  Lets change the name to match the controller that we have